### PR TITLE
Core utils

### PR DESCRIPTION
**Pull request by rete**
BEGINRELEASENOTES
- Implemented memory stat
  - Added MemoryStats structure (unit in Mb)
  - Added OS dependant implementation of `memStats()` for
    - Linux and POSIX
  - Added new dqm4hep-memstats binary
- Fixed json library warning on DQMCore dictionnary generation

ENDRELEASENOTES

### CODE REVIEW

**Review by apingault — DISMISSED** [review]
Looks ok to me

**Comment by rete** [comment]
@apingault Done when TravisCI is happy

File: source/src/Internal.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// -- dqm4hep headers
+#include <dqm4hep/Internal.h>
+
+#include <cstring>
+
+namespace dqm4hep {
+
+  namespace core {
+    
+    // only for unix systems
```
Comment (apingault) [inline source/src/Internal.cc]: Maybe add an #ifdef __linux__ to make sure no other os tries to call it (like APPLE...)

File: source/include/dqm4hep/Internal.h [hunk 2]
```diff
@@ -359,6 +360,43 @@ namespace dqm4hep {
...
+    //-------------------------------------------------------------------------------------------------
+
+    /**
+     *  @brief  Get the current process id
+     */
+    inline int pid() {
+#if _WIN32
+      return GetCurrentProcessId();
+#else
+      return ::getpid();
```
Comment (apingault) [inline source/include/dqm4hep/Internal.h:373]: I was skeptical but this actually works on APPLE too.

**Review by apingault — DISMISSED** [review]
Travis is happy. I'll merge this

File: source/include/dqm4hep/Internal.h (hunk 2 above)
Comment (rete) [reply source/include/dqm4hep/Internal.h:373]: Nice to know !

File: source/src/Internal.cc (hunk 1 above)
Comment (rete) [reply source/src/Internal.cc]: I implemented a fix for that here : 
https://github.com/DQM4HEP/dqm4hep/pull/16

**Comment by rete** [comment]
TravisCI failing because https://github.com/DQM4HEP/dqm4hep/pull/16 not yet merged

File: source/src/Internal.cc (hunk 1 above)
Comment (apingault) [reply source/src/Internal.cc]: Done and tested the fix on OSX.  Waiting for travis now

**Review by apingault — APPROVED** [review]

**Pull request merged**